src/neosophia/agents/utils.py

```diff
@@ -46,6 +46,34 @@ def cprint(*args) -> None:
             print(f"{Colors.BLUE}{var_names[idx].strip()}{Colors.ENDC}: {var}")
 
 
+def summarize_text(
+        text: str,
+        num_sentences: str,
+        num_words: Union[str, int],
+        model: str='gpt-4-0613'):
+    """
+    Summarizes the completed steps given a user command and a list of completed
+    steps.
+
+    Args:
+        command (str): The user input command
+        completed_steps (List[str]): A list of completed steps
+        model (str, optional): The model to be used for chat completion.
+
+    Returns:
+        response (str): The response generated by the chat completion.
+    """
+    prompt = f'Text:\n{text}\n'
+    prompt += sp.SUMMARIZE.replace(
+        '[NUM_SENTENCES]', str(num_sentences)).replace(
+            '[NUM_WORDS]', str(num_words))
```

Comment:
Another wrapping option

```suggestion
    prompt += (
        sp.SUMMARIZE
            .replace('[NUM_SENTENCES]', str(num_sentences))
            .replace('[NUM_WORDS]', str(num_words)))
```